### PR TITLE
[Docs] Agent configuration hierarchy 

### PR DESCRIPTION
**Pull request by natmegs**
This PR updates the Agent configuration docs and splits the existing configuration page into three pages for readability.
* Configuration > General
* Configuration > CLI 
* Configuration > Configuration Files

It follows [this doc](https://docs.google.com/document/d/1Yk_nOIwwoJTHgmyunfm-LTHrKGaWHy0syyWxgAVi3iY/edit) as closely as possible.
There is a lot of opportunity for broken links as these flags and config options are referenced everywhere.
The CLI and config file parameters have been organized into logical(?*) groupings.

*I am not an expert Consul practitioner (UI engineer! Lol) so it is entirely possible that the "logical groupings" are not so logical. Take with a grain of salt.

### CODE REVIEW

File: website/content/docs/agent/config/agent-config-files.mdx
Comment (trujillo-adam) [inline website/content/docs/agent/config/agent-config-files.mdx]:
```suggestion
You can create one or more files to configure the Consul agent on startup. We recommend grouping similar configurations into separate files, such as ACL parameters, to make it easier to manage configuration changes. Using external files may be easier than configuring agents on the command-line when Consul is
```
Have to spread this suggestion out because of the way the breaks in the lines happen.

**Review by trujillo-adam — APPROVED** [review]
Looks great! Made a few suggestions, but I think this will help people consume this information more easily. I am also not a Consul expert, but the segments made sense to me.

**Review by dnephin — COMMENTED** [review]
This looks great, thanks for working on thiS!

I think the only thing this is missing is an update to `redirects.next.js`, so that `agent/options` will redirect to `agent/config`.

File: website/data/docs-nav-data.json
Comment (dnephin) [inline website/data/docs-nav-data.json:781]: What do you think about shortening these paths to this?

```suggestion
            "path": "agent/config/cli-flags"
          },
          {
            "title": "Configuration Reference",
            "path": "agent/config/config-files"
```

We already have agent and config in the path, so we might not need it in the filename as well.

File: website/content/docs/agent/config/agent-config-cli.mdx
Comment (natmegs) [reply website/content/docs/agent/config/agent-config-cli.mdx]: For sure - I added a short `Usage` section which links to those docs. Lmk what you think!

File: website/data/docs-nav-data.json
Comment (natmegs) [reply website/data/docs-nav-data.json]: Makes a lot of sense - updated!

File: CHANGELOG.md
Comment (natmegs) [inline CHANGELOG.md:275]: Is it correct to update references in the changelog.... seems like maybe no? maybe yes?

File: agent/config/runtime.go
Comment (natmegs) [inline agent/config/runtime.go:838]: Found some more references to the old page in some go files.. @dnephin it might be good to get more 👀 on these!

File: CHANGELOG.md
Comment (dnephin) [reply CHANGELOG.md:275]: I think it's a good idea to update these links to point at the new url. Hopefully the redirects work as well, but it doesn't hurt to avoid the redirect.

File: agent/config/runtime.go
Comment (dnephin) [reply agent/config/runtime.go:838]: It's always good to update these, but since we have redirects in place now, I think it's ok if we miss some too.

**Comment by karl-cardenas-coding** [comment]
@natmegs Any chance you can resolve the merge conflicts. I'd love to see this PR get merged :)

**Review by trujillo-adam — APPROVED** [review]
LGTM!

**Comment by jkirschner-hashicorp** [comment]
@natmegs, @trujillo-adam : is anything needed here besides resolving the merge conflicts?

File: website/redirects.next.js
Comment (blake) [inline website/redirects.next.js:1279]:
```suggestion
  {
    source: '/docs/agent/options',
    destination: '/docs/agent/config',
    permanent: true,
  },
```
This is going to break **a lot** of external links outside of Consul.io (Learn, random links scattered across the Internet, etc) that point to config options that used to be on this page. 😞

**Review by jkirschner-hashicorp — CHANGES_REQUESTED** [review]
We should align on @blake 's comment before approve/merge

File: website/redirects.next.js
Comment (dnephin) [reply website/redirects.next.js:1279]: I'm not seeing the diff in that suggestion. Is it a whitespace change?

I think you raise an excellent point. We should strive to make sure those links continue to work.  I think it should be a fair assumption that most links point at refs that were moved to the `config-files` page. There should be fewer links to refs on the `cli-flags` page because there are fewer command line flags, and before most of the time we suggest config files instead.  The inline links edited in this PR seem to confirm that assumption, there are a lot more that point at the new `config-files` page.

I think we can solve this problem by changing the redirect to `destination: /docs/agent/config/config-files`. That should preserve many of the links. Is that what you were suggesting above?


I tested this out with one of the existing redirects. If you click this link, https://www.consul.io/docs/acl/acl-system#acl-policies you should see the browser redirect, and still take you to the ACL Policies section. It worked for me in Chrome, and I hope all other browsers work the same way.

Do you think that is sufficient to preserve most links? A few that link to CLI flags will be broken, but I personally think that's a reasonable cost for the value we get from this change.

**Comment by dnephin** [comment]
I left a suggestion for how to deal with links and redirects above.

The other problem is that there are a lot of merge conflicts :cry: 

Maybe we should try to split this up? With the redirect in place, it should be fine to merge only the changes to `docs/agent/{options, config}` in this PR. We can move all the link updates to one or more follow up PRs, and resolve the conflicts along the way. Let me know if you'd like any help splitting commits, I'd be happy to help.

File: website/redirects.next.js
Comment (blake) [reply website/redirects.next.js:1279]: > I'm not seeing the diff in that suggestion. Is it a whitespace change?
Sorry, I was hoping to suggest an improvement to the redirect that matched on the fragment portion of the URL, and redirected the user to the appropriate page based on whether a `-` was present in the fragment (e.g., `#_raft_trailing_logs`).

I realized that the fragment is not actually part of the URL that is sent to the server, so that isn't even a possible solution.

> I think we can solve this problem by changing the redirect to destination: /docs/agent/config/config-files. That should preserve many of the links. Is that what you were suggesting above?

I think that's a better solution over redirecting to `/config`.

I wonder if its possible to add extra JavaScript to the `/config-files` page that looks at `window.location` and redirects to the user to `/cli-flags` if the fragment/anchor starts with `#_`. 🤔

Comment (karl-cardenas-coding) [reply website/redirects.next.js:1279]: I agree with @dnephin , I think it's worth to take a hit for the increased value we get out of this change.

Comment (jkirschner-hashicorp) [reply website/redirects.next.js:1279]: >Do you think that is sufficient to preserve most links? A few that link to CLI flags will be broken, but I personally think that's a reasonable cost for the value we get from this change.

I also agree with @dnephin, assuming we're sure this is the new page structure we want to build upon long term (where CLI flags and config file parameters are on different pages).

One alternative I thought of which I think is worse, but wanted to raise nonetheless because it would optimize for not breaking anything (at the expense of pointing people to outdated docs) ... We could take advantage of the fact that we have versioned docs and redirect `/docs/agent/options` to an older version of the page (e.g., `/docs/v1.10.x/agent/options`). Newer links wouldn't get redirected (because they would never point to the now non-existent `/docs/agent/options` page). Older links would go to the older page and not be broken (presumably - I haven't tested this).

Comment (karl-cardenas-coding) [reply website/redirects.next.js:1279]: @jkirschner-hashicorp I worry that behavior will generate confusion 😟

Comment (dnephin) [reply website/redirects.next.js:1279]: Good ideas! The javascript redirect could work. We could also put a banner at the top of the new `config-files` page to say something like "looking for command line flags? Go here". And I think we could make the `here` link include the fragment, so they'd go immediately to the right section by clicking that link. That might be slightly easier than automatically redirecting them. I think either is a good option for mitigating the cost. 

> We could take advantage of the fact that we have versioned docs and redirect /docs/agent/options to an older version of the page 

That seems like it would also work. I don't have a strong preference either way, lots of good options.

Comment (karl-cardenas-coding) [reply website/redirects.next.js:1279]: @dnephin that might work with the `<Callouts>` [component](https://react-components-hashicorp.vercel.app/components/callouts)

Comment (trujillo-adam) [reply website/redirects.next.js:1279]: I actually like @jkirschner-hashicorp suggestion about versioning. I've made "breaking changes" in the docs between versions in the past and I most people seem to have found it acceptable. But I'm on board with whatever everyone else decides.

**Comment by trujillo-adam** [comment]
Picking this back up -- was going to checkout this branch and resolve merge the merge conflicts myself, but some of them were in source code files, which I do not know how to address. I don't mind dealing with the .mdx merge conflicts, but I need help to move this forward to completion. It sounds like @dnephin has a solution for addressing the redirects. I'm onboard if everyone else -- but I don't think we need the suggested banner as it will only make sense to us.

**Comment by trujillo-adam** [comment]
@jkirschner-hashicorp @Blake @karl-cardenas-coding - What exactly is holding this up? Are we still weighing broken links vs value of the change? My understanding is that Natalie and Daniel set up matches for redirecting, but we will in all probability still have some broken links. While it would be ideal to redirect 100%, well, you gotta break some eggs . . .  Please let me know what we need to do get this over the finish line.

**Comment by karl-cardenas-coding** [comment]
@trujillo-adam I agree, it's not ideal that we break links, but it will help our practitioners find information easier.  You have my support to proceed.

**Comment by blake** [comment]
@trujillo-adam I think the current behavior in this PR that redirects `/docs/agent/options` to `/docs/agent/config` will be the least intrusive, as it will allow the links to config options to still work.

The links to CLI options will be broken, but hopefully there are fewer of those out in the wild–or at least across our own web properties.

Let me know if you would like help rebasing and resolving merge conflicts with this PR.

**Comment by trujillo-adam** [comment]
> @trujillo-adam I think the current behavior in this PR that redirects `/docs/agent/options` to `/docs/agent/config` will be the least intrusive, as it will allow the links to config options to still work.
> 
> The links to CLI options will be broken, but hopefully there are fewer of those out in the wild–or at least across our own web properties.
> 
> Let me know if you would like help rebasing and resolving merge conflicts with this PR.

Sooo . . . we're good to merge? @jkirschner-hashicorp , does Blake's response satisfy the requested changes? 
@Blake  - I'll probably need to take you up on your offer to help with the conflicts - TBD.

**Comment by jkirschner-hashicorp** [comment]
I agree with both @blake and @karl-cardenas-coding.

Most links within our own docs are to the config file option rather than the CLI option. The longer we leave this branch stale, the more merge conflicts that will emerge, so let's try to get this done in one more concerted push. Thanks for driving this towards completion, @trujillo-adam - it's a valuable improvement to our docs!

**Comment by jkirschner-hashicorp** [comment]
Is this supposed to have 510 commits? That makes it seem like something went wrong with the git-fu for this branch, and that we somehow pulled in a bunch of commits from other places.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/607290.

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit e12a0d083515f99e2e31c024b0f4aa1c732af06e onto `stable-website` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/607290)

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit e12a0d083515f99e2e31c024b0f4aa1c732af06e onto `release/1.11.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/607290)

**Pull request merged**
